bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinWebTargets.java

```diff
@@ -49,13 +61,29 @@
     private String getAirbaseZoneInfoUri;
     private String setAirbaseZoneInfoUri;
 
+    private String uuid;
+    private static HttpClient httpClient;
```

Comment:
Make this non-static and final.

Reply:
I opted to use static to reduce the number of instances of HttpClient within the binding. Using non-static and final, and creating the HttpClient without using the HttpClientFactory will result in the creation of HttpClient multiple times especially each time the discovery process runs. 

There is a warning in HttpClientFactory.java (core code) that says in ALL CAPS:
`DO NOT CREATE NEW CLIENTS FOR EACH REQUEST`

Reply:
Why not just supply the client instance through the constructor?

Reply:
> Why not just supply the client instance through the constructor?

I did that at first. There are two places the instances will need to be created: in the handlerfactory service and in the discovery service. I thought I'd simplify the overall complexity by just creating it inside the DaikinWebTargets. In any case, it will still need to be created, not using the HttpClientFactory.

Should I move back the creation/instantiation into the two services and pass the object through the constructor of DaikinWebTargets?

Reply:
Were we to move the instantiation to the services, there will be two instances of HttpClient created instead of one.

Reply:
You can make your own http client factory service that the discovery service and handler factory service can get a reference to.

Reply:
I would be happy to do this, but I'd like to better understand the problem.

Could you please clarify: are you suggesting to create a separate osgi service that would create the httpclient, and that you're not referring to org.eclipse.smarthome.io.net.http.HttpClientFactory::createHttpClient()?

Could you explain what the reasoning behind this and the benefit over the current approach?

Reply:
Well your current approach of using a static field makes it unclear what controls the lifecycle of the http client. Openhab runs in an osgi environment so it is exepcted that osgi should control the lifecycle of every service and object, either directly or indirectly. Right now if osgi wants to shut down your binding there is no way for it to shut down your static client instance. 

> Could you please clarify: are you suggesting to create a separate osgi service that would create the httpclient

I was suggesting that you create a separate service, since it didn't seem appropriate for your handler factory to have references to the discovery service or vice versa. Having an independent service that could be referenced by both seemed like the best approach.

If you don't want to create a separate osgi service then you could just have your handler factory manually register the http client factory through the bundle context.

> and that you're not referring to org.eclipse.smarthome.io.net.http.HttpClientFactory::createHttpClient()?

No, I am not suggesting to create your own subclass of `org.eclipse.smarthome.io.net.http.HttpClientFactory` since you wouldn't want to have any conflicts in the osgi registry. You would need to create your own factory interface instead.

Reply:
I've created a separate osgi service for the HttpClient.